shell/common/rasterizer.cc

```diff
@@ -270,12 +280,30 @@ RasterStatus Rasterizer::Draw(
       break;
   }
 
-  return draw_result.raster_status;
+  return ToDrawStatus(draw_result.status);
+}
+
+bool Rasterizer::ShouldResubmitFrame(const DoDrawResult& result) {
+  if (result.resubmitted_item) {
+    FML_CHECK(!result.resubmitted_item->layer_tree_tasks.empty());
+    return true;
+  }
+  return false;
 }
 
-bool Rasterizer::ShouldResubmitFrame(const RasterStatus& raster_status) {
-  return raster_status == RasterStatus::kResubmit ||
-         raster_status == RasterStatus::kSkipAndRetry;
+DrawStatus Rasterizer::ToDrawStatus(DoDrawStatus status) {
+  switch (status) {
+    case DoDrawStatus::kEnqueuePipeline:
+      return DrawStatus::kDone;
+    case DoDrawStatus::kNotSetUp:
+      return DrawStatus::kNotSetUp;
+    case DoDrawStatus::kGpuUnavailable:
+      return DrawStatus::kGpuUnavailable;
+    default:
```

Comment:
Please add a case for `DoDrawStatus::kDone` and use `FML_UNREACHABLE` to crash on an unhandled status.

For example:

https://github.com/flutter/engine/blob/d64abdb88ee56888185443dccbffe159d9b18e67/impeller/renderer/backend/vulkan/pipeline_library_vk.cc#L142-L150